scaleway/helpers_lb.go

```diff
@@ -177,41 +168,24 @@ func isPrivateNetworkEqual(a, b interface{}) bool {
 	return false
 }
 
-func newPrivateNetwork(raw map[string]interface{}) *lbSDK.PrivateNetwork {
-	_, pnID, _ := parseZonedID(raw["private_network_id"].(string))
-
-	pn := &lbSDK.PrivateNetwork{PrivateNetworkID: pnID}
-	staticConfig := raw["static_config"]
-	if len(staticConfig.([]interface{})) > 0 {
-		pn.StaticConfig = expandLbPrivateNetworkStaticConfig(staticConfig)
-	} else {
-		pn.DHCPConfig = expandLbPrivateNetworkDHCPConfig(raw["dhcp_config"])
-	}
-
-	return pn
-}
+func privateNetworksCompare(slice1, slice2 []*lbSDK.PrivateNetwork) []*lbSDK.PrivateNetwork {
+	var diff []*lbSDK.PrivateNetwork
 
-func privateNetworksToDetach(pns []*lbSDK.PrivateNetwork, updates interface{}) (map[string]bool, error) {
-	actions := make(map[string]bool, len(pns))
-	configs := make(map[string]*lbSDK.PrivateNetwork, len(pns))
-	// set detached all as default
-	for _, pn := range pns {
-		actions[pn.PrivateNetworkID] = true
-		configs[pn.PrivateNetworkID] = pn
+	m := make(map[string]struct{}, len(slice1))
```

Comment:
What if a private network is removed in the second list?

Reply:
It's covered by the two iterations of privateNetworksCompare, the first call detaches pns not found/changed in the new config and the second attaches pns added/changed. I swap the order of arguments => privateNetworksCompare(pnConfigs, pns) for the detach and privateNetworksCompare(pns, pnConfigs) for the attach